include/dt.h

```diff
@@ -0,0 +1,62 @@
+/***************************************************************************//**
+ *   @file   dt.h
+ *   @brief  Linux device tree like functionality.
+ *   @author Darius Berghe (darius.berghe@analog.com)
+********************************************************************************
+ * Copyright 2021(c) Analog Devices, Inc.
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#ifndef _DT_H_
+#define _DT_H_
+#include <stdint.h>
+
+struct dt_property {
+	const char *name;
+	const int value; // contains the number of elements in array when array is given
```

Comment:
I would prefer if we have something like:
```
struct array { 
        void *data;
        uint32_t len;
}
struct dt_property {
	const char *name;
	enum {
                INTEGER,
                INTEGER_ARRAY,
        } type;
        union {
              struct array array;
              uint32_t     nb;
        } data;
};
```

Reply:
I like my approach because of how similar is to the device tree syntax.
WIth the proposed approach it just doesn't look the same anymore:
`"property", INTEGER_ARRAY, {(int []){1, 2, 3}, 3`
compared to
`"property", 3, (int []){1, 2, 3}`
when the dt looks like
`"property" = < 1 2 3 >

It's normally not good practice to mix the meaning of a variable but in this case it's acceptable for the sake of simplicity, given this is just init configuration stuff and errors are caught right away.